src/app/games/tafl/tests/TaflRules.spec.ts

```diff
@@ -0,0 +1,175 @@
+import { TaflConfig } from '../TaflConfig';
+import { TaflFailure } from '../TaflFailure';
+import { TaflLegalityStatus } from '../TaflLegalityStatus';
+import { TaflMove } from '../TaflMove';
+import { TaflPawn } from '../TaflPawn';
+import { TaflRules, TaflState } from '../TaflRules';
+import { MyTaflMove } from './MyTaflMove.spec';
+import { MyTaflState } from './MyTaflState.spec';
+
+const myTaflConfig: TaflConfig = {
+
+    CASTLE_IS_LEFT_FOR_GOOD: true,
+
+    INVADER: Player.ZERO,
+
+    KING_FAR_FROM_CENTRAL_THRONE_CAN_BE_SANDWICHED: true,
+
+    CENTRAL_THRONE_CAN_SURROUND_KING: true,
+
+    BORDER_CAN_SURROUND_KING: true,
+
+    WIDTH: 7,
+};
+
+class MyTaflRules extends TaflRules<MyTaflMove, MyTaflState> {
+
+    private static singleton: MyTaflRules;
+
+    public static get(): MyTaflRules {
+        if (MyTaflRules.singleton == null) {
+            MyTaflRules.singleton = new MyTaflRules();
+        }
+        return MyTaflRules.singleton;
+    }
+    private constructor() {
+        super(MyTaflState, myTaflConfig, MyTaflMove.from);
+    }
+}
+
+class MyTaflNode extends MGPNode<MyTaflRules, MyTaflMove, MyTaflState> {}
+
+describe('TaflRules', () => {
+
+    let rules: MyTaflRules;
+    let minimaxes: Minimax<TaflMove, TaflState, TaflLegalityStatus>[];
+
+    const _: TaflPawn = TaflPawn.UNOCCUPIED;
+    const O: TaflPawn = TaflPawn.INVADERS;
+    const X: TaflPawn = TaflPawn.DEFENDERS;
+    const A: TaflPawn = TaflPawn.PLAYER_ONE_KING;
+
+    beforeEach(() => {
+        rules = MyTaflRules.get();
+        minimaxes = [
+        ];
+    });
+    describe('getSurroundings', () => {
+        it('Should return neighboorings cases', () => {
+            const startingBoard: Table<TaflPawn> = rules.node.gameState.getCopiedBoard();
+            const { backCoord } =
+                rules.getSurroundings(new Coord(3, 1), Orthogonal.RIGHT, Player.ZERO, startingBoard);
+            expect(backCoord).toEqual(new Coord(4, 1));
+        });
+    });
+    it('should be illegal to move an empty pawn', () => {
+        // Given initial board
+        const state: MyTaflState = MyTaflState.getInitialState();
+
+        // When trying to move an empty piece
+        const move: MyTaflMove = new MyTaflMove(new Coord(0, 1), new Coord(1, 1));
+
+        // Then the move should be illegal
+        const reason: string = RulesFailure.MUST_CHOOSE_PLAYER_PIECE();
+        RulesUtils.expectMoveFailure(rules, state, move, reason);
+    });
+    it('should be illegal to move an opponent pawn', () => {
+        expect(rules.choose(new MyTaflMove(new Coord(4, 2), new Coord(4, 3)))).toBeFalse();
+    });
+    it('should be illegal to land on a pawn', () => {
+        // Given initial board
+        const state: MyTaflState = MyTaflState.getInitialState();
+
+        // When doing a move landing on the ennemy
+        const move: MyTaflMove = new MyTaflMove(new Coord(1, 0), new Coord(1, 3));
+
+        // Then the move should be juged illegal
```

Comment:
should be illegal (le mot correct c'est "judged", mais "should be illegal" c'est juste bon aussi)